test/unit/lib/stack_parser_test.rb

```diff
@@ -0,0 +1,110 @@
+require 'test_helper'
+
+class StackParserTest < ActiveSupport::TestCase
+  class Vulnerable
+  end
+
+  class TestTask < ForemanDeployments::Tasks::BaseDefinition
+  end
+
+  def check_invalid_stack(parser, input)
```

Comment:
nitpick, IMHO it would be nicer to rename it to `assert_stack_invalid` and receive a block and execute, just like `assert_raises`
This way the tests would be a bit clearer:
``` ruby
test 'it raises exception when the stack is not valid yaml hash' do
  assert_ stack_invalid do
    @parser.parse('This is not a valid stack')
  end
end
```